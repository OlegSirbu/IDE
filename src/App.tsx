import React from 'react';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';

import files from './defaultFiles';
import FolderNode from './components/FolderNode';
import MonacoEditor from './components/MonacoEditor';

import {
  parseFileStructure,
} from './utils';
import { useFileEditor } from './hooks/useFileEditor';

const App = () => {
  const fileStructure = parseFileStructure(files);
  const {
    selectedFile,
    handleFileSelect,
    handleEditorChange,
  } = useFileEditor(fileStructure);

  return (
    <Box height="100vh" display="flex">
      <Grid container spacing={2}>
        <Grid item xs={4} md={3}>
          <div data-testid="file-list">
            <FolderNode
              name="root"
              children={fileStructure}
              selectedFilePath={selectedFile.path}
              onSelectFile={handleFileSelect}
            />
          </div>
        </Grid>
        <Grid item xs={8} md={9} data-testid="editor">
          <MonacoEditor
            language={selectedFile.language}
            value={selectedFile.contents}
            onChange={handleEditorChange}
          />
        </Grid>
      </Grid>
    </Box>
  );
};

export default App;
