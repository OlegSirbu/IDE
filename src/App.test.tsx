import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
jest.mock('./components/FolderNode', () => {
  return {
    __esModule: true,
    default: jest
      .fn()
      .mockImplementation(({ onSelectFile }) => (
        <div onClick={() => onSelectFile('path/to/selectedFile.txt')}>
          Mock FolderNode
        </div>
      ))
  };
});



jest.mock('./components/MonacoEditor', () => {
  return {
    __esModule: true,
    default: jest.fn().mockImplementation(() => <div>Mock MonacoEditor</div>)
  };
});

describe('App', () => {
  it('should render correctly ', async () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
