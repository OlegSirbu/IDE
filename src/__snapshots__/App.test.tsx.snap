// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App loads files and selects a file to display in the editor 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-a4s4gt"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-md-4 css-gj1fbr-MuiGrid-root"
      >
        <div
          class="pointer"
        >
          üìÅ root
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-md-8 css-vok43g-MuiGrid-root"
      >
        <section
          style="display: flex; position: relative; text-align: initial; width: 100%; height: 100%;"
        >
          <div
            style="display: flex; height: 100%; width: 100%; justify-content: center; align-items: center;"
          >
            Loading...
          </div>
          <div
            style="width: 100%; display: none;"
          />
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App should render correctly  1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-a4s4gt"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-md-4 css-gj1fbr-MuiGrid-root"
      >
        <div>
          Mock FolderNode
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-md-8 css-vok43g-MuiGrid-root"
      >
        <div>
          Mock MonacoEditor
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
